### PR TITLE
refactor(core/entityTag): Convert entity source components to React

### PR DESCRIPTION
**Pull request by caseyhebebrand**
- Convert `EntitySource` to react with an angular converter
- Set up `ngReact` to reference this component

Future:
- Directly use `EntitySource` in amazon module and remove dependency in `ngReact`

### CODE REVIEW

File: app/scripts/modules/core/src/entityTag/EntitySource.tsx [hunk 1]
```diff
@@ -1,6 +1,76 @@
...
+        execution={execution as IExecution}
+        executionType={executionType}
+        metadata={metadata.value}
+      />
+    ),
+    [executionId],
+  );
+
+  // eslint-disable-next-line
+  console.log(executionType, status, isLoading, metadata);
```
Comment (vigneshm) [inline app/scripts/modules/core/src/entityTag/EntitySource.tsx:54]: You probably forgot to remove it.

File: app/scripts/modules/core/src/entityTag/EntitySource.tsx [hunk 2]
```diff
@@ -1,6 +1,76 @@
...
+    () => (
+      <EntitySourcePopover
+        comments={comments}
+        execution={execution as IExecution}
+        executionType={executionType}
+        metadata={metadata.value}
+      />
+    ),
+    [executionId],
+  );
```
Comment (vigneshm) [inline app/scripts/modules/core/src/entityTag/EntitySource.tsx:51]: Curious on the use of `useCallback` here. I'm wondering if we just define `PopoverContent` as a react element, React may do the right thing.

Also, is `executionId` a good proxy for determining a change in the props supplied to `EntitySourcePopover`?

File: app/scripts/modules/core/src/entityTag/EntitySource.tsx (hunk 1 above)
Comment (caseyhebebrand) [inline app/scripts/modules/core/src/entityTag/EntitySource.tsx:54]:
```suggestion
```

File: app/scripts/modules/core/src/entityTag/EntitySource.tsx [hunk 3]
```diff
@@ -1,6 +1,71 @@
...
+    executionType === 'pipeline' ? { application, executionId, stageId } : { application, taskId: executionId };
+
+  const PopoverContent = () => (
+    <EntitySourcePopover
+      comments={comments}
+      execution={execution as IExecution}
+      executionType={executionType}
+      metadata={metadata.value}
+    />
+  );
```
Comment (vigneshm) [inline app/scripts/modules/core/src/entityTag/EntitySource.tsx:48]: I was previously referring to the following form. Does this work?

```
const PopoverContent = (
  <EntitySourcePopover
      comments={comments}
      execution={execution as IExecution}
      executionType={executionType}
      metadata={metadata.value}
  />
);
```

**Review by vigneshm — APPROVED** [review]

File: app/scripts/modules/core/src/entityTag/EntitySource.tsx (hunk 3 above)
Comment (caseyhebebrand) [reply app/scripts/modules/core/src/entityTag/EntitySource.tsx:48]: The component prop will not accept an element, so I had to make this into a functional component.

**Pull request merged**